File: Arrays/Kids-With-the-Greatest-Number-of-Candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {


        ArrayList<Boolean> bool = new ArrayList<>();
        int mx = Integer.MIN_VALUE;
        for(int i=0; i<candies.length; i++){
            mx = Math.max(mx, candies[i]);
        }
        for(int i=0; i<candies.length; i++){
            int chk = candies[i] + extraCandies;
            if(chk >= mx){
                bool.add(true);
            }else{
                bool.add(false);
            }
        }
        return bool;
    }
}

File: Arrays/Kids-With-the-Greatest-Number-of-Candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:

        ans = []
        max_candies = -sys.maxsize

        for i in candies:
            max_candies = max(max_candies, i)

        for i in candies:
            chk = i + extraCandies
            if chk >= max_candies:
                ans.append(True)
            else:
                ans.append(False)

        return ans
